include/nbl/asset/IAssetManager.h

```diff
@@ -286,30 +294,37 @@ class IAssetManager : public core::IReferenceCounted, public core::QuitSignallin
                     reloadParams.restoreLevels = 0u; // make sure it wont turn into infinite recursion
                     reloadParams.reload = true; // TODO (consider): alternative to this flag: another method in override just to let user choose asset for restore
                 }
-                IAssetLoader::SAssetLoadContext ctx(params, file);
-                auto asset = _override->chooseDefaultAsset(bundle, ctx);
-                if (!asset->isAnyDependencyDummy(restoreLevels - 1u))
-                    return bundle;
 
-                auto reloadBundle = getAssetInHierarchy(_file, _supposedFilename, reloadParams, _hierarchyLevel, _override);
+                auto reloadBundle = getAssetInHierarchy_impl<RestoreWholeBundle>(_file, _supposedFilename, reloadParams, _hierarchyLevel, _override);
 
-                bool restoreSucceeded = _override->handleRestore(std::move(asset), bundle, reloadBundle, restoreLevels);
-                if (!restoreSucceeded) // hm? return empty bundle if restore was requested, but did not succeeded? or just return the bundle? (TODO consider)
-                    return {};
+                if constexpr (RestoreWholeBundle)
```

Comment:
I think you've got your if-statement the wrong way round here

Reply:
oh, right, thanks XD